packages/documentsearch/src/tokens.ts

```diff
@@ -13,26 +14,300 @@ export const ISearchProviderRegistry = new Token<ISearchProviderRegistry>(
   '@jupyterlab/documentsearch:ISearchProviderRegistry'
 );
 
+/**
+ * Filter interface
+ */
+export interface IFilter {
+  /**
+   * Filter title
+   */
+  title: string;
+  /**
+   * Filter description
+   */
+  description: string;
+  /**
+   * Default value
+   */
+  default: boolean;
+  /**
+   * Does the filter support replace?
+   */
+  supportReplace: boolean;
+}
+
+/**
+ * Type of filters
+ *
+ * TODO support additional filter type
+ */
+export interface IFiltersType {
```

Comment:
I would propose called this `IFilters` and removing the word "Type" from the interface name.

One other approach could be making explicit `type` declarations like `BooleanFilter`, etc. and then having `get filters(): Array<BooleanFilter | FooFilter | BarFilter>

But probably just `interface IFilters` is good.

Reply:
(I imagine you wanted to avoid having `IFilter` and `IFilters` but I think the addition of the word `Type` for an `interface` makes it suboptimal)